web/src/components/pages/contribution/contribution.tsx

```diff
@@ -351,7 +351,7 @@ class ContributionPage extends React.Component<ContributionPageProps, State> {
     } = this.props;
     const { selectedPill } = this.state;
 
-    if (isSubmitted && type === 'listen') {
+    if (isSubmitted && type === 'listen' && !user.account) {
```

Comment:
A variable `noUserAccount` has the boolean value of `!user.account` declared below this line (line 358). Could you move that declaration up so we can use that variable instead of `!user.account`?